### PR TITLE
Lookup asynchronously and use PHAssets instead of UIImages

### PR DESCRIPTION
**Pull request by zenangst**

### CODE REVIEW

File: Source/Photos.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      })
+    }
+  }
+
+  static func resolveAsset(asset: PHAsset, size: CGSize = CGSize(width: 720, height: 1280) ,completion: (image: UIImage?) -> Void) {
+    let imageManager = PHImageManager.defaultManager()
+    let requestOptions = PHImageRequestOptions()
+
+    imageManager.requestImageForAsset(asset, targetSize: size, contentMode: PHImageContentMode.AspectFill, options: requestOptions, resultHandler: { image, info in
+      if let info = info where info["PHImageFileUTIKey"] == nil {
```
Comment (zenangst) [inline Source/Photos.swift]: This is a bit dirty but it helps it from not calling the completion twice. This has to do with how `requestImageForAsset` resolves assets.

> For an asynchronous request, Photos may call your result handler block more than once. Photos first calls the block to provide a low-quality image suitable for displaying temporarily while it prepares a high-quality image. (If low-quality image data is immediately available, the first call may occur before the method returns.)

http://stackoverflow.com/questions/27670325/photos-framework-requestimageforasset-returning-two-results-cant-set-image-vi

@hyperoslo/mystore-ios

File: Source/BottomView/StackView.swift [hunk 2]
```diff
@@ -119,11 +120,13 @@ extension ImageStackView {
       return
     }
 
-    let photos = Array(images.suffix(4))
+    let photos = Array(assets.suffix(4))
 
     for (index, view) in views.enumerate() {
       if index <= photos.count - 1 {
-        view.image = photos[index]
+        Photos.resolveAsset(photos[index], size: CGSize(width: Dimensions.imageSize, height: Dimensions.imageSize), completion: { image in
```
Comment (vadymmarkov) [inline Source/BottomView/StackView.swift]: Could be without `completion:`, just open the closure.

File: Source/CameraView/CameraView.swift [hunk 3]
```diff
@@ -191,13 +191,9 @@ class CameraView: UIViewController {
...
           let orientation = self.pictureOrientation()
           let assetsLibrary = ALAssetsLibrary()
-          assetsLibrary.writeImageToSavedPhotosAlbum(image.CGImage, orientation: orientation, completionBlock: nil)
-
-          guard let imageCG = image.CGImage, realOrientation = UIImageOrientation(rawValue: orientation.rawValue) else { return }
-          let rotatedImage = UIImage(CGImage: imageCG,
-            scale: 1.0,
-            orientation: realOrientation)
-          self.delegate?.imageToLibrary(rotatedImage)
+          assetsLibrary.writeImageToSavedPhotosAlbum(image.CGImage, orientation: orientation, completionBlock: { url, error in
```
Comment (vadymmarkov) [inline Source/CameraView/CameraView.swift]: `completionBlock` could be skipped.

File: Source/Photos.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  static func resolveAssets(assets: [PHAsset], size: CGSize = CGSize(width: 720, height: 1280)) -> [UIImage] {
+    let imageManager = PHImageManager.defaultManager()
+    let requestOptions = PHImageRequestOptions()
+
+    var images = [UIImage]()
+
+    requestOptions.synchronous = true
+
+    for asset in assets {
+      imageManager.requestImageForAsset(asset, targetSize: size, contentMode: PHImageContentMode.AspectFill, options: requestOptions, resultHandler: { image, info in
```
Comment (vadymmarkov) [inline Source/Photos.swift]: I think should also work without `resultHandler` :smiley:

File: Source/Photos.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        completion(assets: assets)
+      })
+    }
+  }
+
+  static func resolveAsset(asset: PHAsset, size: CGSize = CGSize(width: 720, height: 1280) ,completion: (image: UIImage?) -> Void) {
+    let imageManager = PHImageManager.defaultManager()
+    let requestOptions = PHImageRequestOptions()
+
+    imageManager.requestImageForAsset(asset, targetSize: size, contentMode: PHImageContentMode.AspectFill, options: requestOptions, resultHandler: { image, info in
```
Comment (vadymmarkov) [inline Source/Photos.swift]: Could drop `resultHandler` and use just `.AspectFill`

**Comment by vadymmarkov** [comment]
@zenangst Made some comments, except of that looks good for me.

**Comment by RamonGilabert** [comment]
It looks good to me!

**Comment by zenangst** [comment]
@vadymmarkov fixed your comments

**Comment by vadymmarkov** [comment]
@zenangst :star2:

**Comment by RamonGilabert** [comment]
Why is this branch restored?

**Comment by zenangst** [comment]
@RamonGilabert I think I push to it by accident

**Comment by mosn** [comment]
Does this mean that we can have [PHAsset] instead of [UIImage]  for `wrapperDidPress` and `doneButtonDidPress` ?

**Comment by onmyway133** [comment]
@mosn As of https://github.com/hyperoslo/ImagePicker/blob/master/Source/ImagePickerController.swift

You can see that it returns `UIImage`

```swift
public protocol ImagePickerDelegate: class {

  func wrapperDidPress(imagePicker: ImagePickerController, images: [UIImage])
  func doneButtonDidPress(imagePicker: ImagePickerController, images: [UIImage])
  func cancelButtonDidPress(imagePicker: ImagePickerController)
}
```

**Pull request merged**